models/articles.models.js

```diff
@@ -53,3 +53,51 @@ exports.updateArticleById = (article_id, inc_votes) => {
     return Promise.reject({ status: 400, msg: "Wrong data type" });
   }
 };
+
+exports.readArticles = (topic) => {
+  let queryStr = `
+  SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, CAST(COUNT(comments.article_id) AS INT) AS comment_count FROM articles
+  LEFT JOIN comments ON comments.article_id = articles.article_id
+  `;
+  const queryValues = [];
+
+  if (topic !== undefined && topic.length === 0) {
+    return Promise.reject({ status: 400, msg: "Invalid topic" });
+  }
+
+  if (topic) {
+    queryStr += `WHERE topic = $1`;
+    queryValues.push(topic);
+  }
+
+  queryStr += `
+  GROUP BY articles.article_id
+  ORDER BY created_at DESC`;
+
+  return db
+    .query(queryStr, queryValues)
+    .then(({ rows, rowCount }) => {
+      if (rowCount === 0) {
+        return Promise.all([
+          rows,
+          db.query(`SELECT * FROM topics WHERE slug=$1`, [topic]),
+        ]);
+      } else {
```

Comment:
this `else` on line 85 doesn't need to exist because if we fall into the `if` statement on line 80 we return out so it will never reach line 86 anyway